tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,680 @@
+            organization=self.org_id, composite=False
+        ).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+
+    @run_only_on('sat')
+    def test_add_synced_docker_repo_to_content_view(self):
+        """@Test: Create and sync a Docker-type repository
+
+        @Assert: A repository is created with a Docker repository
+        and it is synchronized.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(
+            organization=self.org_id
+        ).create_json()['id']
+        repo_id = _create_repository(prod_id)['id']
+
+        task = entities.Repository(id=repo_id).sync()
+        self.assertEqual(u'success', task['result'], task)
+        attrs = entities.Repository(id=repo_id).read_json()
+        self.assertGreaterEqual(attrs[u'content_counts'][u'docker_image'], 1)
+
+        # Create content view and associate docker repo
+        content_view = entities.ContentView(
+            organization=self.org_id, composite=False
+        ).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_add_docker_repo_to_composite_content_view(self, name):
+        """@Test: Add one Docker-type repository to a composite content view
+
+        @Assert: A repository is created with a Docker repository and the
+        product is added to a content view which is then added to a composite
+        content view.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(
+            organization=self.org_id
+        ).create_json()['id']
+
+        repo_id = _create_repository(prod_id, name)['id']
+
+        # Create content view and associate docker repo
+        content_view = entities.ContentView(
+            organization=self.org_id, composite=False
+        ).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+
+        # Publish it...
+        self.assertTrue(_publish_content_view(content_view['id']))
+        # ... and grab its version ID (there should only be one version)
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        version_id = new_attrs['versions'][0]['id']
+
+        # Create composite content view and associate content view to
+        # it
+        comp_content_view = entities.ContentView(
```

Comment:
The only attribute accessed in this dict is `id`.

    comp_content_view_id = entities.ContentView(
        ...
    ).create_json()['id']